src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -307,7 +319,36 @@ export const CohortSelector = ({
       ),
       number: 1,
       onChange,
-      objectSets: [{ values: cohorts, header: 'Saved cohorts' }],
+      objectSets: [
+        {
+          values: cohorts,
+          header: 'Saved cohorts',
+          makeIcon: (value, header) =>
+            h(
+              MenuTrigger,
+              {
+                closeOnClick: true,
+                style: { marginLeft: '1rem' },
+                content: h(Fragment, [
+                  h(
+                    MenuButton,
+                    { 'aria-label': 'Edit cohort', onClick: () => onStateChange(cohortEditorState.new(value)) },
+                    ['Edit']
+                  ),
+                  h(
+                    MenuButton,
+                    {
+                      'aria-label': 'Delete cohort',
+                      onClick: () => updateCohorts((cohorts) => _.without([value], cohorts)),
```

Comment:
I think this is the same as
```suggestion
                      onClick: () => updateCohorts(_.without([value])),
```